src/map/ai/controllers/player_controller.cpp

```diff
@@ -110,6 +111,12 @@ bool CPlayerController::Ability(uint16 targid, uint16 abilityid)
     auto* PChar = static_cast<CCharEntity*>(POwner);
     if (PChar->PAI->CanChangeState())
     {
+        CAbility* PAbility = ability::GetAbility(abilityid);
+        if (PChar->PRecastContainer->HasRecast(RECAST_ABILITY, PAbility->getRecastId(), PAbility->getRecastTime()))
```

Comment:
I believe `GetAbility` can fail and return `nullptr`, can you add a check for that?

Weaponskill does the same thing:
```cpp
        if (PWeaponSkill == nullptr)
        {
            PChar->pushPacket(new CMessageBasicPacket(PChar, PChar, 0, 0, MSGBASIC_CANNOT_USE_WS));
            return false;
        }
        ```